lib/compiler-arguments.js

```diff
@@ -0,0 +1,168 @@
+            _.forEach(this.possibleArguments, (obj, argKey) => {
+                for (var argIdx in excludeUsedArguments) {
+                    if (this.match(argKey, excludeUsedArguments[argIdx])) return;
+                }
+                possibleArguments[argKey] = obj;
+            });
+        } else {
+            possibleArguments = this.possibleArguments;
+        }
+
+        let arr = _.pairs(possibleArguments);
+        arr.sort((a, b) => {
+            if ((a[1].timesused === 0) && (b[1].timesused === 0)) {
+                // prefer optimization flags or standard if statistics are not available
+                if (a[1].description.includes("optimization")) {
+                    return -1;
+                } else if (b[1].description.includes("optimization")) {
+                    return 1;
+                } else if (a[1].description.includes("std")) {
+                    return -1;
+                } else if (b[1].description.includes("std")) {
+                    return 1;
+                }
+            } else {
+                return (b[1].timesused - a[1].timesused);
+            }
+        });
+        arr = _.first(arr, this.maxPopularArguments);
+        return _.object(arr);
+    }
+
+    populateOptions(options) {
+        this.possibleArguments = options;
+    }
+
+    match(documentedOption, givenOption) {
+        if (documentedOption.includes("<number>") || documentedOption.includes("<n>")) {
+            const numre = /[0-9]*$/i;
+            if (documentedOption.indexOf(givenOption.replace(numre, "")) === 0) {
+                return documentedOption;
+            }
+        }
+
+        if (documentedOption.includes("=")) {
+            const idx = documentedOption.indexOf("=");
+            if (givenOption.indexOf("=") === idx) {
+                if (documentedOption.substr(0, idx) === givenOption.substr(0, idx)) {
+                    return documentedOption;
+                }
+            }
+        }
+
+        if (documentedOption.includes(":")) {
+            const idx = documentedOption.indexOf(":");
+            if (givenOption.indexOf(":") === idx) {
+                if (documentedOption.substr(0, idx) === givenOption.substr(0, idx)) {
+                    return documentedOption;
+                }
+            }
+        }
+
+        if (documentedOption.includes("[")) {
+            const idx = documentedOption.indexOf("[") - 1;
+            if (documentedOption.indexOf(givenOption.substr(0, idx)) === 0) {
+                return documentedOption;
+            }
+        }
+        
+        if (documentedOption.indexOf(givenOption) === 0) {
+            return documentedOption;
+        }
+
+        return false;
+    }
+
+    addOptionToStatistics(option, timesUsed) {
+        if (!timesUsed) timesUsed = 1;
+
+        const possibleKeys = _.compact(
+            _.keys(this.possibleArguments).map((val) => this.match(val, option))
+        );
+
+        if (possibleKeys.length > 0) {
```

Comment:
I think this check is unnecessary? The `forEach` handles the "it's empty" case. Saves an indentation :)

Reply:
ah, you're right